### PR TITLE
feat(integrations): New `SysExitIntegration`

### LINKED ISSUE #2636: SystemExit not capturable
### How do you use Sentry?

Sentry Saas (sentry.io)

### Version

1.39.2

### Steps to Reproduce

SystemExit (raised manually or through sys.exit) are not captured by sentry_sdk.
We are not using any wsgi framework: sentry is used to monitor script (running directly through `python myscript.py`)

I found no way to specify that I didn't want them to be ignored after looking at the doc and the codebase. I only found [this issue ](https://github.com/getsentry/sentry-python/issues/379) which had them handled in WSGI integration.

I understand some people might not want these to be captured, but not having the option to capture them seems like a bug to me.

We can manually trigger sentry when sys.exit is called in our code, but it'd be quite painful having to patch all potential packages raising sys.exit (like argparse).

minimal working example:
Python 3.9.16
```
import atexit
import sys

import sentry_sdk


def before_send(event, hint):
    print("in before_send")
    return event


sentry_sdk.init(
    dsn="obfuscated",
    attach_stacktrace=True,  # attach stacktrace to logs in addition to errors
    environment="local",
    before_send=before_send,
    # doc: https://docs.sentry.io/platforms/python/configuration/options/
)
atexit.register(lambda: sentry_sdk.flush())  # removes error logs on shutdown


def myfunction():
    exit = 2 # tried 1 & 2

    print(f"running sys.exit({exit})")
    sys.exit(exit)  # not triggering before_send

    # print(f"raise systemexit {exit}")
    # raise SystemExit(exit) # not triggering before_send

    # raise BaseException("test maxime")  # triggers before_send

    raise Exception("test maxime")  # triggers before_send


myfunction()
```

### Expected Result

All the above are captured by sentry (`sys.exit(1)`, `sys.exit(2)`, `raise SystemExit(1)`, `raise SystemExit(2)`)

### Actual Result

None are captured by sentry (`sys.exit(1)`, `sys.exit(2)`, `raise SystemExit(1)`, `raise SystemExit(2)`) : 
```
(py39) ➜  repo git:(master) ✗ python test.py
running sys.exit(2)
(py39) ➜  repo git:(master) ✗ python test.py
running sys.exit(1)
(py39) ➜  repo git:(master) ✗ python test.py
raise systemexit 1
(py39) ➜  repo git:(master) ✗ python test.py
in before_send
Traceback (most recent call last):
(...)
BaseException
```

### PR DESCRIPTION
**Pull request by szokeasaurusrex**
The `SysExitIntegration` reports `SystemExit` exceptions raised by calls made to `sys.exit` with a value indicating unsuccessful program termination – that is, any value other than `0` or `None`. Optionally, by setting `capture_successful_exits=True`, the `SysExitIntegration` can also report `SystemExit` exceptions resulting from `sys.exit` calls with successful values.

You need to manually enable this integration if you wish to use it.

Closes #2636

Docs issue: https://github.com/getsentry/sentry-docs/issues/11043

### CODE REVIEW

**Comment by szokeasaurusrex** [comment]
I still need to write docs for this

File: sentry_sdk/integrations/sys_exit.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+from sentry_sdk.integrations import Integration
+from sentry_sdk._types import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+    from typing import NoReturn, Union
+
+
+class SysExitIntegration(Integration):
+    """Captures sys.exit calls and sends them as an events to Sentry.
```
Comment (antonpirker) [inline sentry_sdk/integrations/sys_exit.py:18]:
```suggestion
    """Captures sys.exit calls and sends them as events to Sentry.
```

**Review by antonpirker — APPROVED** [review]
Very cool!

We need to add this to the docs though

**Comment by szokeasaurusrex** [comment]
Docs issue: https://github.com/getsentry/sentry-docs/issues/11043

**Comment by szokeasaurusrex** [comment]
Gonna go ahead with merging this, the docs PR is in review

File: sentry_sdk/integrations/sys_exit.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    identifier = "sys_exit"
+
+    def __init__(self, *, capture_successful_exits=False):
+        # type: (bool) -> None
+        self._capture_successful_exits = capture_successful_exits
+
+    @staticmethod
+    def setup_once():
+        # type: () -> None
+        SysExitIntegration._patch_sys_exit()
```
Comment (glensc) [inline sentry_sdk/integrations/sys_exit.py:37]: you could use classmethod instead. then you would not need to know name of current class.

```py
    @classmethod
    def setup_once(cls):
        # type: () -> None
        cls._patch_sys_exit()
```

**Pull request merged**